aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+    DEPTH = ds.createVariable(varname='DEPTH', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    UCURqc = ds.createVariable(varname='UCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    VCURqc = ds.createVariable(varname='VCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    WCURqc = ds.createVariable(varname='WCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    DEPTHqc = ds.createVariable(varname='DEPTH_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    TIME = ds.createVariable(varname='TIME', datatype='double', zlib=True, dimensions=('OBSERVATION'))
+    instrument_index = ds.createVariable(varname='instrument_index', datatype='int', zlib=True, dimensions=('OBSERVATION'))
+
+    source_file = ds.createVariable(varname='source_file', datatype='S256', dimensions=('INSTRUMENT'))
+    instrument_id = ds.createVariable(varname='instrument_id', datatype='S256', dimensions=('INSTRUMENT'))
+    LATITUDE = ds.createVariable(varname='LATITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    LONGITUDE = ds.createVariable(varname='LONGITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    NOMINAL_DEPTH = ds.createVariable(varname='NOMINAL_DEPTH', datatype='float', dimensions=('INSTRUMENT'))
+
+    ## main loop
+    for index, file in enumerate(files_to_agg):
+        print(index)
+        with xr.open_dataset(file) as nc:
+
+            ## in water data only
+            nc = in_water(nc)
+
+            start = sum(varlen_list[:index + 1])
+            end = sum(varlen_list[:index + 2])
+            n_cells = get_nvalues(nc)[1]
+            UCUR[start:end] = get_varvalues(nc, 'UCUR')
+            UCURqc[start:end] = get_varvalues(nc, 'UCUR_quality_control')
+            VCUR[start:end] = get_varvalues(nc, 'VCUR')
+            VCURqc[start:end] = get_varvalues(nc, 'VCUR_quality_control')
+            if 'WCUR' in nc.data_vars:
+                WCUR[start:end] = get_varvalues(nc, 'WCUR')
+                WCURqc[start:end] = get_varvalues(nc, 'WCUR_quality_control')
+            else:
+                WCUR[start:end] = np.full(varlen_list[index], np.nan)
+                WCURqc[start:end] = np.full(varlen_list[index], np.nan)
+            ##calculate depth
+            if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+                DEPTH[start:end] = (nc.DEPTH + nc.HEIGHT_ABOVE_SENSOR.T).values.flatten()
+                DEPTHqc[start:end] = np.array(n_cells * [nc.DEPTH_quality_control.values]).flatten()
+            else:
+                DEPTH[start:end] = nc.DEPTH.values
+                DEPTHqc[start:end] = nc.DEPTH_quality_control.values
+            ## set TIME and instrument index
+            TIME[start:end] = (np.repeat(get_varvalues(nc, 'TIME'), n_cells) - epoch) / one_day
```

Comment:
The expression inside the parenthesis is a bit long. Since this is an assignement operation, I would do without a big parenthesis 
```
TIME[start:end] = np.repeat(...)/one_day - epoch/one_day
```
The intent of normalization is much better (and should be slightly quicker).